lib/krb5/build_auth.c

```diff
@@ -73,10 +74,93 @@ make_etypelist(krb5_context context,
 	krb5_abortx(context, "internal error in ASN.1 encoder");
     free_EtypeList(&etypes);
 
-    ret = _krb5_add_1auth_data(context,
-                               KRB5_AUTHDATA_GSS_API_ETYPE_NEGOTIATION, &e, 0,
-                               auth_data);
+    ade.ad_type = KRB5_AUTHDATA_GSS_API_ETYPE_NEGOTIATION;
+    ade.ad_data = e;
+
+    ret = add_AuthorizationData(auth_data, &ade);
+
     krb5_data_free(&e);
+
+    return ret;
+}
+
+/* HACK: temp */
+static void
+encode_le_uint32(uint32_t n, uint8_t *p)
```

Comment:
by inline in my earlier comment, I meant just put this code directly in `add_ad_options()` – it's a bit ugly but at least it's a common design pattern. Or use the `krb5_storage` API as you did originally.